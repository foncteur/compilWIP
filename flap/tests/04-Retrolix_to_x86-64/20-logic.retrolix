external printf

def print_one_int ()
local a1:
  a1: a1 <- copy %rdi;
  l00: %rdi <- copy "%d\n";
  l01: %rsi <- copy a1;
  l02: %rax <- copy 0;
  l03: printf();
  l04: ret;
end

globals (x)
  l11: %rdi <- and 1, 0;
  l12: print_one_int();
  l13: %rdi <- and 1, 1;
  l14: print_one_int();
  l15: %rdi <- and 0, 1;
  l16: print_one_int();
  l17: %rdi <- and 0, 0;
  l18: print_one_int();
  l19: exit;
end