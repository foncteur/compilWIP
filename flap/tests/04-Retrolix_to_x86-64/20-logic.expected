Process exited with status exited(0).
STDOUT:
0
1
0
0

STDERR:



